x-pack/plugins/observability_solution/observability_ai_assistant_app/public/functions/visualize_esql.tsx

```diff
@@ -259,150 +259,156 @@ export function VisualizeESQL({
+                          : i18n.translate(
+                              'xpack.observabilityAiAssistant.lensESQLFunction.displayTable',
+                              {
+                                defaultMessage: 'Display results',
+                              }
+                            )
+                      }
+                    />
+                  </EuiToolTip>
+                </EuiFlexItem>
                 <EuiToolTip
-                  content={i18n.translate('xpack.observabilityAiAssistant.lensESQLFunction.save', {
-                    defaultMessage: 'Save visualization',
+                  content={i18n.translate('xpack.observabilityAiAssistant.lensESQLFunction.edit', {
+                    defaultMessage: 'Edit visualization',
                   })}
                 >
                   <EuiButtonIcon
                     size="xs"
-                    iconType="save"
-                    onClick={() => setIsSaveModalOpen(true)}
-                    data-test-subj="observabilityAiAssistantLensESQLSaveButton"
+                    iconType="pencil"
+                    onClick={() => {
+                      chatFlyoutSecondSlotHandler?.setVisibility?.(true);
+                      if (triggerOptions) {
+                        uiActions.getTrigger('IN_APP_EMBEDDABLE_EDIT_TRIGGER').exec(triggerOptions);
+                      }
+                    }}
+                    data-test-subj="observabilityAiAssistantLensESQLEditButton"
                     aria-label={i18n.translate(
-                      'xpack.observabilityAiAssistant.lensESQLFunction.save',
+                      'xpack.observabilityAiAssistant.lensESQLFunction.edit',
                       {
-                        defaultMessage: 'Save visualization',
+                        defaultMessage: 'Edit visualization',
                       }
                     )}
                   />
                 </EuiToolTip>
-              </EuiFlexItem>
-            </EuiFlexGroup>
-          </EuiFlexItem>
-          <EuiFlexItem data-test-subj={visualizationComponentDataTestSubj}>
-            {isTableVisible ? (
-              <ESQLDataGrid
-                rows={rows}
-                columns={columns}
-                dataView={dataViewAsync.value}
-                query={{ esql: query }}
-                flyoutType="overlay"
-                isTableView
-              />
-            ) : (
-              <lens.EmbeddableComponent
-                {...lensInput}
-                style={{
-                  height: 240,
-                }}
-                onLoad={onLoad}
-              />
-            )}
+                <EuiFlexItem grow={false}>
+                  <EuiToolTip
+                    content={i18n.translate(
+                      'xpack.observabilityAiAssistant.lensESQLFunction.save',
+                      {
+                        defaultMessage: 'Save visualization',
+                      }
+                    )}
```

Comment:
these labels are duplicated I think? maybe move them to a constant in this file?

Reply:
They are, yes I will do

Reply:
Done https://github.com/elastic/kibana/pull/184923/commits/58f2c3832485dad9f5d25ca044aeef8bfdf5bac8 (also moved the edit visualization ones)